index.js

```diff
@@ -373,62 +373,86 @@
   ];
 
   //  ValidDate :: Type
-  $.ValidDate = $.NullaryType(
+  $.ValidDate = NullaryType(
     'sanctuary-def/ValidDate',
     function(x) { return $.Date.test(x) && !isNaN(x.valueOf()); }
   );
 
+  //  PositiveNumber :: Type
+  $.PositiveNumber = NullaryType(
+    'sanctuary-def/PositiveNumber',
+    function(x) { return $.Number.test(x) && x > 0; }
```

Comment:
The shadowing point makes sense.

> The more types the environment contains, the more types may be listed in error messages

This is going to be a problem whenever environments with large hierarchies are used. Is your expectation that users will be restrictive in the types they use in their environments? I bet many applications could get away with excluding `$.Number` and including only `$.PositiveInteger`.

Do you think we might include other, pre-built, environments that provide other convenient groupings?